learntools/machine_learning/nbs/ch3-testing.ipynb

```diff
@@ -197,99 +572,185 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Step 4: Specify Model\n",
-    "Create a `DecisionTreeRegressorModel` and save it iowa_model. Ensure you've done the relevant import from sklearn to run this command."
+    "## Step 3: Specify and Fit Model\n",
+    "Create a `DecisionTreeRegressor` and save it iowa_model. Ensure you've done the relevant import from sklearn to run this command.\n",
+    "\n",
+    "Then fit the model you just created using the data in `X` and `y` that you saved above."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 9,
    "metadata": {
     "collapsed": true
    },
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Incorrect\n"
+     ]
+    },
+    {
+     "data": {
+      "text/markdown": [
+       "<span style=\"color:#cc3333\">Incorrect:</span> You have not fit the model."
+      ],
+      "text/plain": [
+       "Incorrect: You have not fit the model."
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
    "source": [
-    "#iowa_model = _\n",
-    "step_4.check()"
+    "from sklearn.tree import DecisionTreeRegressor\n",
+    "iowa_model = DecisionTreeRegressor(random_state=1)\n",
+    "print(\"Incorrect\")\n",
+    "step_3.check()"
```

Comment:
This would also help with another goal on my wishlist: improving the reproducibility of the outputs of notebooks we're keeping in version control, to keep diffs as cruftless as possible. Rerunning some of these notebooks I see silly diffs like...

```
-      "First in-sample predictions: [ 208500.  181500.  223500.  140000.  250000.]\n",
+      "First in-sample predictions: [208500. 181500. 223500. 140000. 250000.]\n",
```

Which are probably just a result of us having different versions of pandas locally installed.

Reply:
I wonder how annoying it would be to start testing everything from docker using Kaggle's docker image.